pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -568,33 +580,37 @@ protected void internalCreatePartitionedTopic(AsyncResponse asyncResponse, int n
             return;
         }
         validateNamespaceOperationAsync(topicName.getNamespaceObject(), NamespaceOperation.CREATE_TOPIC)
-                .thenRun(() -> {
-                    Policies policies = null;
-                    try {
-                        policies = getNamespacePolicies(namespaceName);
-                    } catch (RestException e) {
-                        if (e.getResponse().getStatus() != Status.NOT_FOUND.getStatusCode()) {
-                            throw e;
+                .thenCompose((__) -> getNamespacePoliciesAsync(namespaceName).exceptionally(ex -> {
+                    Throwable unwrapped = FutureUtil.unwrapCompletionException(ex);
+                    if (unwrapped instanceof RestException re) {
+                        if (re.getResponse().getStatus() == Status.NOT_FOUND.getStatusCode()) {
+                            return null;
                         }
                     }
-
+                    throw new CompletionException(unwrapped);
```

Comment:
could use `throw FutureUtil.wrapToCompletionException(ex);` instead. There wouldn't be a need to create a new `CompletionException` when it's already a CompletionException.